### PR TITLE
Error loading

### LINKED ISSUE #14: As a user, I should see a loading message or error message on the main page if my data is not there

### LINKED ISSUE #15: When a user clicks more info, a loading message should appear when data is not there

### LINKED ISSUE #17: As a user, I should not be able to add duplicate trips to my 'future trips'

### LINKED ISSUE #19: An error should be rendered if data is unable to be fetched from the server
- add a .catch and error messages/components into all components with fetch calls

### PR DESCRIPTION
**Pull request by ashton-huxtable**
# Description
Error and loading states added for components

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring

# How Has This Been Tested?
- [ ] open localHost
- [ ] Cypress
- [ ] dev tools
- [x] React Server 

# Checklist:
- [x] My code follows the Turing's guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] No console error messages

### CODE REVIEW

File: src/components/Home/Home.js [hunk 1]
```diff
@@ -8,20 +8,22 @@ import StatePicker from '../StatePicker/StatePicker';
...
   const [error, setError] = useState('')
-  const [loading, setLoading] = useState('')
 
   useEffect(() =>{ 
     getParksByActivities()
     .then(data => {
      const hikeParks = filterParksForHiking(data)
       setParks(hikeParks)
     })
+    .catch(err => setError('Something went wrong. Please try again later'))
```
Comment (clairefields15) [inline src/components/Home/Home.js:26]: In getParksByActivities, do you pass your data through an error checking fn? Your catch won't necessarily catch all errors.

File: src/components/Home/Home.js [hunk 2]
```diff
@@ -36,13 +38,20 @@ const Home = () => {
   return(
     <main>
       <Switch>
+        {error && <h2 className='error'>{error}</h2>}
```
Comment (clairefields15) [inline src/components/Home/Home.js]: You might want to do !!error to show the error and only show your other paths if !error, but maybe that's just a weird thing I've been doing

File: src/components/Home/Home.js [hunk 3]
```diff
@@ -36,13 +38,20 @@ const Home = () => {
...
         <Route exact path='/' render={() => <StatePicker selectState={selectState}/> } />
-        <Route path='/parksByState' render={() => <ParkList parksByState={parksByState}/>}/>
-        <Route path='/futureVisits' render={() => <FutureVisits futureTrips={futureTrips}/>} />
+        {(!parksByState.length && !error) ?
+          <h2 className='error'>Loading park list...</h2> :
```
Comment (clairefields15) [inline src/components/Home/Home.js:44]: Nice!

File: src/components/Home/Home.js [hunk 4]
```diff
@@ -36,13 +38,20 @@ const Home = () => {
...
+          <Route path='/parksByState' render={() => <ParkList parksByState={parksByState}/>}/>
+        }
         <Route path='/details/:parkCode' render={({match}) => {
           const selectedPark = parksByState.find(park => park.parkCode === match.params.parkCode)
           return <ParkDetails selectedPark={selectedPark} addToFutureTrips={addToFutureTrips} />
         }}/>
+        {!futureTrips.length ?
+          <h2 className='error'>No trips saved! Return to home to start planning!</h2> :
+          <Route path='/futureVisits' render={() => <FutureVisits futureTrips={futureTrips}/>} />
+        }
```
Comment (clairefields15) [inline src/components/Home/Home.js:54]: Oh nice, I've never seen the path="*" but that makes sense. If it's in a switch, can also just say <Route> and don't give it any path to catch everything.

File: src/components/ParkDetails/ParkDetails.js [hunk 5]
```diff
@@ -6,31 +6,37 @@ import ParkActivities from '../ParkActivities/ParkActivities';
...
+  const [error, setError] = useState('')
 
   const getDetails = () => {
     getParkDetailsData(selectedPark.parkCode)
       .then(data => {
         const details = filterParkDetails(data)
         setParkDetails(details)
       })
-  }
+      .catch(err => setError('Something went wrong. Please try again later!'))
```
Comment (clairefields15) [inline src/components/ParkDetails/ParkDetails.js:23]: Same question as above, do you catch more specific errors as part of your fetch call?

File: src/components/ParkDetails/ParkDetails.js [hunk 6]
```diff
@@ -6,31 +6,37 @@ import ParkActivities from '../ParkActivities/ParkActivities';
...
+    {(!Object.keys(parkDetails).length && error) ? <h2 className='error'>{error}</h2> :
       <main className='park-details'>
         <Link to='/parksByState'>
           <button>go back</button>
         </Link>
         {
-          !Object.keys(parkDetails).length ?
-          <h2>loading......</h2> :
+          (!Object.keys(parkDetails).length && !error) ?
+          <h2>Loading park details...</h2> :
```
Comment (clairefields15) [inline src/components/ParkDetails/ParkDetails.js:39]: I mean, I think you really needed those three extra ... to really get the point across 😆

File: src/components/SunriseSunset/SunriseSunset.js [hunk 7]
```diff
@@ -61,16 +64,19 @@ const SunriseSunset = ({latitude, longitude, image, fullName, addToFutureTrips,
...
-          <img className='riseset-img'src={sunset} alt='sunset icon'/>
-          <h5>Sunset:  {sunRiseSet.sunset}</h5>
-        </section>
-      </div>
+      {(!Object.keys(sunRiseSet).length && error) ?
+        <h2 className='error'>{error}</h2> :
+        <div className='sunrise-sunset'>
+          <section className='sunrise'>
+            <img className='riseset-img' src={sunrise} alt='sunrise icon'/>
+            <h5>Sunrise:  {sunRiseSet.sunrise}</h5>
```
Comment (clairefields15) [inline src/components/SunriseSunset/SunriseSunset.js:72]: You might get a lighthouse error for this, non-consecutive headings (h2 => h5)

File: src/components/Error/NotFound.js [hunk 8]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import './Error.css'
```
Comment (clairefields15) [inline src/components/Error/NotFound.js:2]: Would it be better to dynamically pass in a "message" as props to this error component, and have one of those messages be 404: page not found?

**Review by clairefields15 — COMMENTED** [review]
Looks good! Just a few notes but it looks pretty clean. I didn't check out your fetch call file but be sure that you are catching all the errors with a checkForErrors fn before json-ing your response

File: src/components/Home/Home.js (hunk 1 above)
Comment (ashton-huxtable) [reply src/components/Home/Home.js:26]: Added a check for error in my apiCalls file!

**Pull request merged**